### PR TITLE
SPLAT-1171: add installer platform-external workflow

### PR DESCRIPTION
**Pull request by rvanderp3**
Add installer presubmit jobs which allow platform-external to be tested with and without an external CCM.

### CODE REVIEW

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-none-vsphere

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-none-vsphere

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-none-vsphere

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-none-vsphere

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-none-vsphere

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-none-vsphere

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm

File: ci-operator/step-registry/upi/conf/vsphere/platform-none/upi-conf-vsphere-platform-none-commands.sh [hunk 1]
```diff
@@ -9,9 +9,17 @@ base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
...
+  if [[ $EXTERNAL_CCM != "enabled" ]]; then
   PLATFORM_TYPE="external:
     platformName: ${PLATFORM_NAME}
   "
+  else
+  PLATFORM_TYPE="external:
+    platformName: ${PLATFORM_NAME}
+    cloudControllerManager: External
+  "
+  fi
```
Comment (mtulio) [inline ci-operator/step-registry/upi/conf/vsphere/platform-none/upi-conf-vsphere-platform-none-commands.sh:22]: Nit:
```suggestion
  PLATFORM_TYPE="external:
    platformName: ${PLATFORM_NAME}
  "
  if [[ $EXTERNAL_CCM == "enabled" ]]; then
    PLATFORM_TYPE="external:
    platformName: ${PLATFORM_NAME}
    cloudControllerManager: External
  "
  fi
```

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-platform-external-no-ccm pull-ci-openshift-installer-master-e2e-vsphere-platform-external-ccm

**Comment by mtulio** [comment]
@rvanderp3 I will hold this PR quickly to prevent saving baseline artifacts from installer executions. The PR fixing it is in progress https://github.com/openshift/release/pull/43383
/hold

**Comment by mtulio** [comment]
@rvanderp3 #43383 merged. feel free to drop the `/hold` when you are happy. For me:
/lgtm

**Pull request closed without merging**